version_check_test.go

```diff
@@ -0,0 +1,57 @@
+package main
+
+import (
+	"net/http"
+	"net/http/httptest"
+	"testing"
+)
+
+func TestVersionCheck(t *testing.T) {
+	defer expectReleaseVersion("1.1.3", 302)()
+
+	v, err := getLatestReleaseVersion()
+	if err != nil {
+		t.Fatal(err)
+	}
+	if x, v := "1.1.3", v; x != v {
+		t.Errorf("expected release version to be %q, was %q", x, v)
+	}
+}
+
+func TestValidateVersionWithErr(t *testing.T) {
+	baseVersion := "1.1.11"
+	errorCases := []string{"1.0.0", "1.1.10", "1.0.11", "1.1.10-dev"}
+
+	defer expectReleaseVersion(baseVersion, 302)()
+
+	for _, version := range errorCases {
+		PHRASEAPP_CLIENT_VERSION = version
+		err := validateVersionWithErr()
+		if err == nil {
+			t.Errorf("expected an error but got %s >= %s", version, baseVersion)
+		}
+	}
+
+	noErrorCases := []string{"1.1.11", "1.1.12", "1.2.10", "2.0.0"}
+	for _, version := range noErrorCases {
+		PHRASEAPP_CLIENT_VERSION = version
+		err := validateVersionWithErr()
+		if err != nil {
+			t.Errorf("expected no error but got %s, for %s < %s", err, PHRASEAPP_CLIENT_VERSION, baseVersion)
+		}
+	}
+}
```

Comment:
it is also goot to reset the PHRASEAPP_CLIENT_VERSION to the original value when that one test is done